modules/instrument_builder/jsx/react.app.js

```diff
@@ -0,0 +1,485 @@
+TabPane = React.createClass({
+    render: function() {
+        var classList = "tab-pane";
+        if(this.props.Active) {
+            classList += " active"
+        }
+        return (
+            <div className={classList} id={this.props.TabId}>
+                <h1>{this.props.Title}</h1>
+                <br />
+                {this.props.children}
+            </div>
+            );
+    }
+});
+LoadPane = React.createClass({
+	getInitialState: function() {
+	 	return {
+	 		alert: ''
+	 	};
+	},
+	chooseFile: function (e) {
+		var value = e.target.files[0];
+		this.setState(function(state){
+			return {
+				file: value,
+				alert: ''
+			};
+		});
+	},
+	setAlert: function (type) {
+		this.setState(function(state){
+			return {
+				alert: type
+			};
+		});
+	},
+	resetAlert: function () {
+		this.setState(function(state){
+			return {
+				alert: ''
+			};
+		});
+	},
+	loadFile: function () {
+		var callback = {
+				success: this.props.loadCallback,
+				error: this.setAlert
+			};
+		Instrument.load(this.state.file, callback);
+	},
+	render: function () {
+		var spanDownStyle = {
+				display: 'none'
+			},
+			alert = '';
+		switch (this.state.alert) {
+			case 'success':
+				alert = (
+					<div className="alert alert-success alert-dismissible" role="alert">
+						  <button type="button" className="close" onClick={this.resetAlert}><span aria-hidden="true">&times;</span></button>
+						  <strong>Success!</strong> Instrument Loaded
+					</div>
+				)
+				break;
+			case 'typeError':
+				alert = (
+					<div className="alert alert-danger alert-dismissible" role="alert">
+						  <button type="button" className="close" onClick={this.resetAlert}><span aria-hidden="true">&times;</span></button>
+						  <strong>Error!</strong> Wrong file format
+					</div>
+				)
+				break;
+		}
+		return (
+			<TabPane Title="Load Instrument"
+                TabId={this.props.TabId}>
+                	<div className="col-sm-4 col-xs-12">
+                		{alert}
+						<input className="fileUpload"
+							   type="file" id="instfile"
+			            	   onChange={this.chooseFile}
+						/>
+			            <br />
+			            <input className="btn btn-primary"
+			            	   type="button" id="load"
+			            	   value="Load Instrument"
+			            	   onClick={this.loadFile}
+			            />
+			        </div>
+			</TabPane>
+		);
+	}
+});
+SavePane = React.createClass({
+	getInitialState: function() {
+	 	return {
+	 		fileName: '',
+	 		instrumentName: ''
+	 	};
+	},
+	loadState: function(newState) {
+		this.setState(function(state){
+			return {
+				fileName: newState.fileName,
+				instrumentName: newState.instrumentName
+			};
+		});
+	},
```

Comment:
I don't understand what you're trying to do with these setState wrapper functions. setState usually takes an object as a parameter. Why are you passing a function that returns an array inside of a closure that doesn't use the parameters?